src/query.rs

```diff
@@ -22,17 +35,55 @@ use crate::{
 /// # use rs_ecs::*;
 /// let mut world = World::new();
 ///
-/// let entity = world.alloc();
-/// world.insert(entity, (0_i32, true));
+/// let entity1 = world.alloc();
+/// world.insert(entity1, (42_i32, 23_u32, 1.0_f32));
 ///
-/// let entity = world.alloc();
-/// world.insert(entity, (42_i32, 23_u32, 1.0_f32));
+/// let entity2 = world.alloc();
+/// world.insert(entity2, (0_i32, true));
 ///
-/// let mut query = Query::<(&i32, &mut bool)>::new();
-/// for (i, b) in query.borrow(&world).iter() {
-///     *b = *i > 10;
+/// let mut query = Query::<(&mut i32, Option<(&u32, &mut f32)>)>::new();
+/// for (i, opt) in query.borrow(&world).iter() {
+///     if let Some((u, f)) = opt {
+///         *i += *u as i32;
+///         *f += 1.0
+///     } else {
+///         *i -= 1;
+///     }
 /// }
+///
+/// assert_eq!(*world.get::<i32>(entity1).unwrap(), 65);
+/// assert_eq!(*world.get::<f32>(entity1).unwrap(), 2.0);
+/// assert_eq!(*world.get::<i32>(entity2).unwrap(), -1);
 /// ```
+///
+/// Use of a prepared query that is stored and reused for optimization:
+///
+/// ```
+/// # use rs_ecs::*;
+/// #[derive(Default)]
+/// struct System {
+///     query: Query<(&'static i32, &'static mut bool)>
+/// }
+///
+/// impl System {
+///     pub fn update(&mut self, world: &World) {
+///         for (i, b) in self.query.borrow(world).iter() {
+///             *b = *i >= 0;
+///         }
+///     }
+/// }
+///
+/// fn main() {
+///     let mut world = World::new();
+///
+///     let entity = world.alloc();
+///     world.insert(entity, (0_i32, true));
+///
+///     let mut system = System::default();
+///     system.update(&world);
```

Comment:
Maybe put this into some kind of loop to really reuse the system?

Reply:
Yes, that makes sense.

Reply:
Will merge and amend the loop. (Just doing the tag and publish before calling it a day.)